docs/Documentation/Software_Tools/spack.md

```diff
@@ -0,0 +1,302 @@
+|`spack deactivate --all <parent>`	            |deactivates all extensions of parent (e.g., <python>) | 
+
+
+## Modules
+
+Spack can auto-create environment modulefiles for the packages that it builds, both in Tcl for “environment modules” per se, and in Lua for Lmod. 
+Auto-creation includes each dependency and option permutation, which can lead to excessive quantities of modulefiles. 
+Spack also uses the package hash as part of the modulefile name, which can be somewhat disconcerting to users. 
+These default behaviors can be treated in the active modules.yaml file, as well as practices used for support.
+Tcl modulefiles are created in `$SPACK_ROOT/share/spack/modules` by default, and the equivalent Lmod location is `$SPACK_ROOT/share/spack/lmod`. 
+Only Tcl modules are created by default. 
+You can modify the active modules.yaml file in the following ways to affect some example behaviors:
+
+### To turn Lmod module creation on,
+
+```
+modules:
+    enable:
+        - tcl
+        - lmod 
+```
+
+### To change the modulefile naming pattern,
+
+```
+modules:
+    tcl:
+        naming_scheme: ‘{name}/{version}/{compiler.name}-{compiler.version}
+```
+
+would achieve the Eagle naming scheme. 
+### To remove default variable settings in the modulefile, e.g., CPATH,
+
+```
+modules:
+    tcl:
+        all:
+            filter:
+                environment_blacklist: [‘CPATH’]
+```
+
+Note that this would affect Tcl modulefiles only; if Spack also creates Lmod files, those would still contain default CPATH modification behavior.
+
+### To prevent certain modulefiles from being built, you can whitelist and blacklist.
+
+```
+modules:
+    tcl:
+        whitelist: [‘gcc’]
+        blacklist: [‘%gcc@4.8.5’]
+```
+
+This would create modules for all versions of GCC built using the system compiler, but not for the system compiler itself.
+There are a great many further behaviors that can be changed, see https://spack.readthedocs.io/en/latest/module_file_support.html#modules for more.
+
+For general user support, it is not a bad idea to keep the modules that are publicly visible separate from the collection that Spack auto-generates. This involves some manual copying, but is generally not onerous as all rpaths are included in Spack-built binaries (i.e., you don’t have to worry about satisfying library dependencies for Spack applications with an auto-built module, since library paths are hard-coded into the application binaries). This separation also frees one from accepting Spack’s verbose coding formats within modulefiles, should you decide to maintain certain modulefiles another way.
+
+## Configuration
+
+Spack uses hierarchical customization files. 
+Every package is a Python class, and inherits from the top-level class Package. 
+Depending on the degree of site customization, you may want to fork the Spack repo to create your own customized Spack package.
+There are 4 levels of configuration. In order of increasing precedence,
```

Comment:
This list does not format correctly, may need a line break before the list begins.